### DISCUSSION TITLE
What is the use of factory keyword while creating name constructor in model class?

### DISCUSSION THREAD

**Discussion by ramanajees** [community/community#28070] [Mobile] [answered]
What is the use of factory keyword while creating name constructor in model class?

**Comment by sthakrishnakumar** [comment, marked as answer]
Hello @ramanajees ,
factory keyword is used in a constructor when you don't necessarily want a constructor to create a new instance of your class.
It might be helpful when you hold instances of  class in memory and don't want to create a new one each time.

**Comment by surajadkhari** [comment]
A factory constructor is a constructor that can be used when you don't necessarily want a constructor to create a new instance of your class.

This might be useful if you hold instances of your class in memory and don't want to create a new one each time (or if the operation of creating an instance is costly).
